app/assets/javascripts/map.js

```diff
@@ -1,152 +1,165 @@
+    origin: this.start,
+    destination: this.end,
+    waypoints: this.waypts,
+    optimizeWaypoints: true,
+    travelMode: google.maps.TravelMode.DRIVING
+  };
+
+  directionsService.route(request, function(response, status){
+    if (status == google.maps.DirectionsStatus.OK) {
+
+      decoder = new PolylineDecoder()
+      route.polyline = decoder.decodePolyline(response);
+      polygon = new Polygon(route.polyline)
+      route.displayRoute(response, true)
+    }
+  });
+};
 
-          if (status == google.maps.DirectionsStatus.OK) {
-            directionsDisplay.setDirections(response);
 
-            var polyline = decodePolyline(response);
-            var polygonGeoJsonObject = createPolygonFromPolyline(polyline);
+Route.prototype.displayRoute = function(response, drawMe){
+  drawMe = typeof drawMe !== 'undefined' ? drawMe : true;
 
-// comment this line out to prevent drawing of polygon
-            var polygonCoords = processPolygonCoordsIntoLatLong(polygonGeoJsonObject)
-          }
-        });
-      }
+  directionsDisplay.setDirections(response);
 
-//------------------------------------------------------------
+  if (drawMe) {
+    drawer = new Drawer();
+    drawer.draw(polygon.geoJson);
+  };
+}
 
-  // Define the LatLng coordinates for the polygon's path.
-  function processPolygonCoordsIntoLatLong(polygonGeoJsonObject) {
-    var polygonCoords = createLatLongObjects(polygonGeoJsonObject);
-    drawPolygon(polygonCoords)
-    return polygonCoords
 
-  }
+//----polyline decoder-----
 
-  function createLatLongObjects(geoJsonObject){
-    var latLongArray = []
-    var coordArray = geoJsonObject.features[0].geometry.coordinates[0]
+var PolylineDecoder = function(response){}
 
-    console.log(coordArray)
+PolylineDecoder.prototype.decodePolyline = function(response){
+  var coord_array = polyline.decode(response.routes[0].overview_polyline);
+  return coord_array.map(function(coordinate) {
+    return [coordinate[1], coordinate[0]];
+  })
+}
+
+//-------Polygon!------------
+
+var Polygon = function(polyline){
+  this.geoJson = this.createGeoJsonFromPolyline(polyline)
+  this.searchWithin(this.geoJson)
+}
+
+Polygon.prototype.createGeoJsonFromPolyline = function(polyline) {
+  var line = {
+    type:"Feature",
+    geometry:{
+      type:"LineString",
+      coordinates: polyline
+     }
+   }
+
+  var polygon = turf.buffer(line, 25, 'miles')
+  return polygon
+}
 
-    coordArray.forEach(function(coord){
-      console.log(coord)
-      latLongArray.push(new google.maps.LatLng(coord[1], coord[0]))
-      return latLongArray
-    })
+Polygon.prototype.searchWithin = function(polygon){
+  $.ajax({
+  url: '/maps/search',
+  type: 'post',
+  beforeSend: function(xhr) { xhr.setRequestHeader('X-CSRF-Token', $('meta[name=csrf-token]').attr('content'))},
+  contentType: "application/json",
+  dataType: "json",
+  data: JSON.stringify(polygon),
+})
+  .success(function(response){
+    loadMarkers(response.attractions)
+  })
```

Comment:
I'd advise you write a fail condition, even if it's just `alert('bad things')` its helpful in debugging.